### ISSUE TITLE
`react/require-default-props` false positive with destructured properties

### ISSUE THREAD

**Issue by herndlm** [yannickcr/eslint-plugin-react#3237]
Since 7.29.x (still working fine with 7.28.x) I get the following errors
```
  4:3  error  propType "legacyApi" is not required, but has no corresponding defaultProps declaration  react/require-default-props
  6:3  error  propType "title" is not required, but has no corresponding defaultProps declaration      react/require-default-props
```

for

```ts
import React from 'react';

export interface ChartOptions {
  legacyApi?: boolean,
  query: string,
  title?: string,
  type: string,
}

export default ({
  legacyApi = false, query, title = '', type,
}: ChartOptions) => (
  <div>
    <span>{{ legacyApi }}</span>
    <span>{{ query }}</span>
    <span>{{ title }}</span>
    <span>{{ type }}</span>
  </div>
);
```

Is this a bug or am I doing something wrong here? If the latter than sorry for the spam. I can't figure out what I'm doing wrong here.

**Comment by ljharb** [comment]
This is a correct warning; function defaults aren’t the same as defaultProps. Your component needs a defaulProps property, which is an object containing the defaults.

**Comment by herndlm** [comment]
Oh.
Ok, thx for the fast response and sorry for the noise then :)
